fedora_messaging/_session.py

```diff
@@ -495,7 +495,7 @@ def consume(self, callback, bindings=None, queues=None, exchanges=None):
                     " or a function."
                 )
             self._consumer_callback = cb_obj
-        elif inspect.isfunction(callback):
+        elif inspect.isfunction(callback) or inspect.ismethod(callback):
```

Comment:
Looking at docs for `callable`:
```
    Note that classes are callable, as are instances of classes with a
    __call__() method.
```

The only problem I see is when someone passes a class as a callback which doesn't implement `__call__`.

```python
In [1]: class C:
   ...:     pass
   ...: 
   ...: 

In [2]: callable(C)
Out[2]: True

In [3]: callback = C

In [4]: message = {}

In [5]: callback(message)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-5d33cb4c2f38> in <module>()
----> 1 callback(message)

TypeError: C() takes no arguments
```

Reply:
Indeed. The ``if inspect.isclass(callback)`` should handle this particular case, but in general at the moment we don't validate that the callback really does accept only 1 positional argument. It'd be nice to do that and I assume Python has a way to inspect that, but it wouldn't be a regression in functionality since we don't handle this nicely now.